src/components/ProjectSettingsTabs/PluginsTab/getFormSchema.ts

```diff
@@ -0,0 +1,189 @@
+import { CardFieldTemplate } from "components/SpruceForm/FieldTemplates";
+import widgets from "components/SpruceForm/Widgets";
+import { hiddenIf, placeholderIf, radioBoxOptions } from "../utils";
+import { FormState } from "./types";
+
+export const getFormSchema = (
+  useRepoSettings: boolean,
+  repoData?: FormState,
+  formData?: FormDataProps
+): {
+  fields: Record<string, Field>;
+  schema: SpruceFormProps["schema"];
+  uiSchema: SpruceFormProps["uiSchema"];
+} => ({
+  fields: {},
+  schema: {
+    type: "object" as "object",
+    properties: {
+      performanceSettings: {
+        type: "object" as "object",
+        title: "Performance Plugins",
+        properties: {
+          perfEnabled: {
+            type: ["boolean", "null"],
+            title: "",
+            oneOf: radioBoxOptions(
+              ["Enabled", "Disabled"],
+              repoData?.performanceSettings?.perfEnabled
+            ),
+          },
+        },
+      },
+      buildBaronSettings: {
+        type: "object" as "object",
+        title: "Build Baron and Task Annotations",
+        properties: {
+          taskAnnotationSettings: {
+            title: "",
+            type: "object" as "object",
+            properties: {
+              jiraCustomFields: {
+                type: "array" as "array",
+                title: "Custom Jira Fields",
+                items: {
+                  type: "object" as "object",
+                  properties: {
+                    field: {
+                      type: "string" as "string",
+                      title: "Field",
+                    },
+                    displayText: {
+                      type: "string" as "string",
+                      title: "Display Text",
+                    },
+                  },
+                },
+              },
+            },
+          },
+          useBuildBaron: {
+            type: "boolean" as "boolean",
+            oneOf: radioBoxOptions(
+              [
+                "Build Baron Ticket Search and Create",
+                "Custom Ticket Creation",
+              ],
+              undefined
+            ),
+          },
+          ticketSearchProjects: {
+            type: "array" as "array",
+            title: "Ticket Search Projects",
+            items: {
+              type: "object" as "object",
+              properties: {
+                searchProject: {
+                  type: "string" as "string",
+                  title: "Search Project",
+                },
+              },
+            },
+          },
+          ticketCreateProject: {
+            type: "object" as "object",
+            title: "Ticket Create Project",
+            properties: {
+              createProject: {
+                type: "string" as "string",
+                title: "",
+              },
+            },
+          },
+
+          fileTicketWebhook: {
+            type: "object" as "object",
+            title: "Custom Ticket Creation",
```

Comment:
Likewise, if one of Endpoint/Secret is populated, should the other be required?

Reply:
I don't personally think so. Especially since nobody is using it yet and we don't know what their setup is (and if they'll even use a secret). A secret without an endpoint would be nonsensical, but since we are assuming that it'll be used by intelligent users, I don't know if we need to block saving due to it, especially since it won't harm anything. I'm open to suggestions though.

Reply:
I'd personally prefer to block it; I think it's out pattern to block users from doing nonsensical things -- a user can't save a variable without a key for example, I believe (cc @sophstad to verify).

Reply:
Correct, blocking save is the tack we've generally been taking so would make sense to continue here. This could be done using a custom validation function like we did for the variables page [here](https://github.com/evergreen-ci/spruce/blob/main/src/components/ProjectSettingsTabs/VariablesTab/VariablesTab.tsx#L53)